chips/imxrt10xx/src/gpio.rs

```diff
@@ -732,1230 +535,133 @@ impl ClockInterface for PortClock {
-                port.registers.gdir.modify(GDIR::GDIR13.val(mode as u32));
-            }
-            0b001110 => {
-                port.registers.gdir.modify(GDIR::GDIR14.val(mode as u32));
-            }
-            0b001111 => {
-                port.registers.gdir.modify(GDIR::GDIR15.val(mode as u32));
-            }
-            0b010000 => {
-                port.registers.gdir.modify(GDIR::GDIR16.val(mode as u32));
-            }
-            0b010001 => {
-                port.registers.gdir.modify(GDIR::GDIR17.val(mode as u32));
-            }
-            0b010010 => {
-                port.registers.gdir.modify(GDIR::GDIR18.val(mode as u32));
-            }
-            0b010011 => {
-                port.registers.gdir.modify(GDIR::GDIR19.val(mode as u32));
-            }
-            0b010100 => {
-                port.registers.gdir.modify(GDIR::GDIR20.val(mode as u32));
-            }
-            0b010101 => {
-                port.registers.gdir.modify(GDIR::GDIR21.val(mode as u32));
-            }
-            0b010110 => {
-                port.registers.gdir.modify(GDIR::GDIR22.val(mode as u32));
-            }
-            0b010111 => {
-                port.registers.gdir.modify(GDIR::GDIR23.val(mode as u32));
-            }
-            0b011000 => {
-                port.registers.gdir.modify(GDIR::GDIR24.val(mode as u32));
-            }
-            0b011001 => {
-                port.registers.gdir.modify(GDIR::GDIR25.val(mode as u32));
-            }
-            0b011010 => {
-                port.registers.gdir.modify(GDIR::GDIR26.val(mode as u32));
-            }
-            0b011011 => {
-                port.registers.gdir.modify(GDIR::GDIR27.val(mode as u32));
-            }
-            0b011100 => {
-                port.registers.gdir.modify(GDIR::GDIR28.val(mode as u32));
-            }
-            0b011101 => {
-                port.registers.gdir.modify(GDIR::GDIR29.val(mode as u32));
-            }
-            0b011110 => {
-                port.registers.gdir.modify(GDIR::GDIR30.val(mode as u32));
-            }
-            0b011111 => {
-                port.registers.gdir.modify(GDIR::GDIR31.val(mode as u32));
-            }
-            _ => {}
+    fn get_mode(&self) -> Mode {
+        if self.registers.gdir.get().is_bit_set(self.offset) {
+            Mode::Output
+        } else {
+            Mode::Input
         }
     }
 
-    pub fn get_pinid(&self) -> PinId {
-        self.pinid
+    fn set_mode(&self, mode: Mode) {
+        let gdir = self.registers.gdir.get();
+        let gdir = match mode {
+            Mode::Input => gdir.clear_bit(self.offset),
+            Mode::Output => gdir.set_bit(self.offset),
+        };
+        self.registers.gdir.set(gdir);
     }
 
     fn set_output_high(&self) {
-        let port = self.pinid.get_port();
-
-        match self.pinid.get_pin_number() {
-            0b000000 => {
-                port.registers.dr.write(DR::DR0::SET);
```

Comment:
From my understanding, a `write` usage will set all non-specified fields to zero. These calls on `dr` may need to be a modify, otherwise setting one pin high will set all other pins low.

Alternatively, write a high bit `dr_set`, which will set the pin high.